### PR TITLE
MRG, STY: Many style fixes

### PR DESCRIPTION
**Pull request by larsoner**
~~WIP until~~ https://github.com/numpy/numpydoc/pull/238 is merged, ~~then I'll change~~ so I've  changed our code to use `numpy/numpydoc/master`.

Basically the Pandas folks had their own docstring validation code that they are now contributing to NumpyDoc, and it found a bunch of inconsistencies in our code. This fixes many of them, but leaves the following for later when someone is bored (there are hundreds and they are annoying):
```
    'PR06',  # Parameter type should use "str" instead of "string"
    'PR08',  # Parameter description should start with a capital letter
    'PR09',  # Parameter description should finish with "."
    'RT04',  # Return value description should start with a capital letter
    'RT05',  # Return value description should finish with "."
```
The only downside I can see is that we lose our "parameter description length" check, but this is easy enough to complain about in the review process.

### CODE REVIEW

File: mne/decoding/transformer.py [hunk 1]
```diff
@@ -81,7 +81,7 @@ class Scaler(TransformerMixin, BaseEstimator):
     info : instance of Info | None
         The measurement info. Only necessary if ``scalings`` is a dict or
         None.
-    scalings : dict, string, default None.
+    scalings : dict, string, default None
```
Comment (drammock) [inline mne/decoding/transformer.py]: shouldn't this one be 
```
scalings : dict | str | None
```

Comment (larsoner) [reply mne/decoding/transformer.py]: yes but let's work on that in a separate PR, there are probably many such instances in our code base. I'll make a comment above about this

**Comment by larsoner** [comment]
Okay this PR is ready to go now that the upstream NumpyDoc PR has been merged

**Comment by agramfort** [comment]
thx @larsoner

**Pull request merged**